tests/main_test.py

```diff
@@ -163,10 +164,186 @@ def test_old_baseline_ignored_with_update_flag(
+                    },
+                    {
+                        "name": "PrivateKeyDetector",
+                    },
+                ],
+                '--no-base64-string-scan --no-keyword-scan',
+                [
+                    {
+                        "name": "PrivateKeyDetector",
+                    },
+                ],
+            ),
+            (  # all plugins
+                [
+                    {
+                        "base64_limit": 1.5,
+                        "name": "Base64HighEntropyString",
+                    },
+                ],
+                '--use-all-plugins',
+                [
+                    {
+                        "name": "AWSKeyDetector",
+                    },
+                    {
+                        "base64_limit": 4.5,
+                        "name": "Base64HighEntropyString",
+                    },
+                    {
+                        "name": "BasicAuthDetector",
+                    },
+                    {
+                        "hex_limit": 3,
+                        "name": "HexHighEntropyString",
+                    },
+                    {
+                        "name": "KeywordDetector",
+                    },
+                    {
+                        "name": "PrivateKeyDetector",
+                    },
+                ],
+            ),
+            (  # remove some plugins from all plugins
+                [
+                    {
+                        "base64_limit": 4.5,
+                        "name": "Base64HighEntropyString",
+                    },
+                ],
+
+                '--use-all-plugins --no-base64-string-scan --no-private-key-scan',
+                [
+                    {
+                        "name": "AWSKeyDetector",
+                    },
+                    {
+                        "name": "BasicAuthDetector",
+                    },
+                    {
+                        "hex_limit": 3,
+                        "name": "HexHighEntropyString",
+                    },
+                    {
+                        "name": "KeywordDetector",
+                    },
+                ],
+            ),
+            (  # use same plugin list from baseline
+                [
+                    {
+                        "base64_limit": 3.5,
+                        "name": "Base64HighEntropyString",
+                    },
+                    {
+                        "name": "PrivateKeyDetector",
+                    },
+                ],
+                '',
+                [
+                    {
+                        "base64_limit": 3.5,
+                        "name": "Base64HighEntropyString",
+                    },
+                    {
+                        "name": "PrivateKeyDetector",
+                    },
+                ],
+            ),
+            (  # ignore overwriten option from CLI when not using --use-all-plugins
```

Comment:
Hmm. If you **explicitly** override it by providing the CLI argument, even without `--use-all-plugins`, wouldn't you expect CLI to take precedence over a configured baseline?

Example use case: user wants to see if a decrease in sensitivity results in a large drop in noise.

```
$ detect-secrets scan --update .secrets.baseline --base64-limit 5
$ git diff .secrets.baseline
```

Reply:
I debated about that, it might have some edge cases. For example, if the baseline does not have the base64 scan, do we ignore the limit? or add the base64 scan in? So I chose to let the user on purposely use the plugin, then allow adjusting. How would you suggest to adjust?

Reply:
It's updated to support these cases now. New test cases added.